auto_submit/lib/service/revert_request_validation_service.dart

```diff
@@ -0,0 +1,172 @@
+import 'package:auto_submit/validations/validation.dart';
+import 'package:github/github.dart' as github;
+import 'package:retry/retry.dart';
+
+class RevertRequestValidationService extends ValidationService {
+  RevertRequestValidationService(Config config, {RetryOptions? retryOptions})
+      : super(config, retryOptions: retryOptions) {
+    /// Validates a PR marked with the reverts label.
+    approverService = ApproverService(config);
+  }
+
+  ApproverService? approverService;
+  Revert? revertValidation;
+
+  /// Processes a pub/sub message associated with PullRequest event.
+  Future<void> processMessage(github.PullRequest messagePullRequest, String ackId, PubSub pubsub) async {
+    if (await shouldProcess(messagePullRequest)) {
+      await processRevertRequest(
+        config: config,
+        result: await getNewestPullRequestInfo(config, messagePullRequest),
+        messagePullRequest: messagePullRequest,
+        ackId: ackId,
+        pubsub: pubsub,
+      );
+    } else {
+      log.info('Should not process ${messagePullRequest.toJson()}, and ack the message.');
+      await pubsub.acknowledge('auto-submit-queue-sub', ackId);
+    }
+  }
+
+  Future<bool> shouldProcess(github.PullRequest pullRequest) async {
+    final github.RepositorySlug slug = pullRequest.base!.repo!.slug();
+    final GithubService githubService = await config.createGithubService(slug);
+    final github.PullRequest currentPullRequest = await githubService.getPullRequest(slug, pullRequest.number!);
+    final List<String> labelNames = (currentPullRequest.labels as List<github.IssueLabel>)
+        .map<String>((github.IssueLabel labelMap) => labelMap.name)
+        .toList();
+
+    final RepositoryConfiguration repositoryConfiguration = await config.getRepositoryConfiguration(slug);
+
+    if (currentPullRequest.state == 'open' && labelNames.contains(Config.kRevertLabel)) {
+      // TODO (ricardoamador) this will not make sense now that reverts happen from closed.
+      // we can open the pull request but who do we assign it to? The initiating author?
+      if (!repositoryConfiguration.supportNoReviewReverts) {
+        log.info(
+          'Cannot allow revert request (${slug.fullName}/${pullRequest.number}) without review. Processing as regular pull request.',
+        );
+        final int issueNumber = currentPullRequest.number!;
+        // Remove the revert label and add the autosubmit label.
+        await githubService.removeLabel(slug, issueNumber, Config.kRevertLabel);
+        await githubService.addLabels(slug, issueNumber, [Config.kAutosubmitLabel]);
+        return false;
+      }
+      return true;
+    }
+    return false;
+  }
+
+  /// TODO this becomes validate to determine if the pR status is good to proceed.
+  /// Checks if a pullRequest is still open and with autosubmit label before trying to process it.
+  Future<ProcessMethod> processPullRequestMethod(github.PullRequest pullRequest) async {
```

Comment:
Is this method still needed?

Reply:
Ah thank you! It is not needed. I remember removing this but maybe I forgot with some conflicts I had to address.